### PR TITLE
Update server.py

### PR DESCRIPTION
**Pull request by fawziabd**
error handling

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -66,11 +66,17 @@ def process_data(data, connection: socket.socket, user, index):
...
+        # Check command
+        if (len(data) != 1):
+            return "400 Invalid command format"
+        
         return database.get_balance(user[0])
     elif data[0] == "deposit":
         # Check command
         if (len(data) != 2):
             return "400 Invalid command format"
+        elif not data[1].isdigit():
```
Comment (jweb89) [inline server.py]: Thanks for catching! Can you add this to the if statement as or since it's the same return

File: server.py [hunk 2]
```diff
@@ -66,11 +66,17 @@ def process_data(data, connection: socket.socket, user, index):
     elif data[0] == "list":
         return database.list_stocks(user) if not user[3] else database.list_stocks_root()
     elif data[0] == "balance":
+        # Check command
+        if (len(data) != 1):
```
Comment (jweb89) [inline server.py:70]: Do we need these? Wouldn't it not enter the elif data[0] == 'balance'?

**Review by jweb89 — CHANGES_REQUESTED** [review]

File: server.py (hunk 2 above)
Comment (fawziabd) [reply server.py:70]: its within the elif data[0] == 'balance' block? Just to show an error if anything is typed after 'balance'

Comment (jweb89) [reply server.py:70]: Ahhh I see. Cool

File: server.py (hunk 1 above)
Comment (jweb89) [reply server.py]: Can you add this check to the if statement on line 76 and then I'll go ahead and merge it

**Review by jweb89 — APPROVED** [review]

**Pull request merged**